Kudu.Services.Web/App_Start/NinjectServices.cs

```diff
@@ -148,6 +148,9 @@ private static void RegisterServices(IKernel kernel)
 
             EnsureDotNetCoreEnvironmentVariable(environment);
 
+            // fix up invalid d:\home\site\deployments\settings.xml
+            EnsureValidDeploymentXmlSettings(environment);
```

Comment:
Why do this during startup as opposed to when we actually need to open the file?

Reply:
The file opening is done in https://github.com/davidfowl/XmlSettings.  Fixing it there will change the XmlSettings behavior.  Besides, I am not sure if XmlSettings should delete the corrupted setting file.

I was thinking about doing it at DeploymentSettingsManager; however, it encapsulates many setting providers.   I am afraid it may look weird try/catch and delete some file.

So I landed on a simplest solution specific to settings.xml.

Reply:
Ok, fair enough, I see how you got there.

As an aside, that tiny NuGet package is a bit silly, and we really should just have the code in Kudu.